Lib/test/test_uuid.py

```diff
@@ -497,61 +506,66 @@ def check_node(self, node, requires=None, network=False):
         self.check_node(node, 'ifconfig', True)
 
     @unittest.skipUnless(os.name == 'posix', 'requires Posix')
     def test_ip_getnode(self):
-        node = uuid._ip_getnode()
+        node = self.uuid._ip_getnode()
         self.check_node(node, 'ip', True)
 
     @unittest.skipUnless(os.name == 'posix', 'requires Posix')
     def test_arp_getnode(self):
-        node = uuid._arp_getnode()
+        node = self.uuid._arp_getnode()
         self.check_node(node, 'arp', True)
 
     @unittest.skipUnless(os.name == 'posix', 'requires Posix')
     def test_lanscan_getnode(self):
-        node = uuid._lanscan_getnode()
+        node = self.uuid._lanscan_getnode()
         self.check_node(node, 'lanscan', True)
 
     @unittest.skipUnless(os.name == 'posix', 'requires Posix')
     def test_netstat_getnode(self):
-        node = uuid._netstat_getnode()
+        node = self.uuid._netstat_getnode()
         self.check_node(node, 'netstat', True)
 
     @unittest.skipUnless(os.name == 'nt', 'requires Windows')
     def test_ipconfig_getnode(self):
-        node = uuid._ipconfig_getnode()
+        node = self.uuid._ipconfig_getnode()
         self.check_node(node, 'ipconfig', True)
 
     @unittest.skipUnless(importable('win32wnet'), 'requires win32wnet')
     @unittest.skipUnless(importable('netbios'), 'requires netbios')
     def test_netbios_getnode(self):
-        node = uuid._netbios_getnode()
+        node = self.uuid._netbios_getnode()
         self.check_node(node, network=True)
 
     def test_random_getnode(self):
-        node = uuid._random_getnode()
+        node = self.uuid._random_getnode()
         # Least significant bit of first octet must be set.
         self.assertTrue(node & 0x010000000000, '%012x' % node)
         self.check_node(node)
 
     @unittest.skipUnless(os.name == 'posix', 'requires Posix')
-    @unittest.skipUnless(importable('ctypes'), 'requires ctypes')
-    def test_unixdll_getnode(self):
-        try: # Issues 1481, 3581: _uuid_generate_time() might be None.
-            node = uuid._unixdll_getnode()
-        except TypeError:
-            self.skipTest('requires uuid_generate_time')
-        self.check_node(node)
+    def test_unix_getnode(self):
+        if not importable('_uuid') and not importable('ctypes'):
+            self.skipTest("neither _uuid extension nor ctypes available")
+        self.check_node(self.uuid._unix_getnode(), 'unix')
 
     @unittest.skipUnless(os.name == 'nt', 'requires Windows')
     @unittest.skipUnless(importable('ctypes'), 'requires ctypes')
     def test_windll_getnode(self):
-        node = uuid._windll_getnode()
+        node = self.uuid._windll_getnode()
         self.check_node(node)
 
 
+class TestInternalsWithoutExtModule(BaseTestInternals, unittest.TestCase):
+    uuid = py_uuid
+
+@unittest.skipUnless(c_uuid, 'requires the C _uuid module')
+class TestInternalsWithExtModule(BaseTestInternals, unittest.TestCase):
+    uuid = c_uuid
```

Comment:
I don't see the point of testing all uuid.*getnode() functions with and without _uuid. Basically, it's the same test run twice, no?

Only getnode() and uuid1() should be tested with and without _uuid, no?

Reply:
> Only getnode() and uuid1() should be tested with and without _uuid, no?

Probably. I didn't want to modify the tests too much but I'll take a look.